apis/Google.Longrunning/Google.Longrunning/Operation.cs

```diff
@@ -0,0 +1,329 @@
+        // but we could at least ignore Done entirely in this code...
+
+        /// <summary>
+        /// Whether the operation has completed, where "complete" can include "failed".
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        public bool IsCompleted => RpcMessage.Done;
+
+        /// <summary>
+        /// Whether the operation has completed with a failure.
+        /// </summary>
+        public bool IsFaulted => RpcMessage.ResultCase == Operation.ResultOneofCase.Error;
+
+        /// <summary>
+        /// The error associated with the operation, as an <see cref="OperationFailedException"/>, or <c>null</c>
+        /// if the operation is not in an error state (either because it completed successfully, or because it
+        /// has not yet completed).
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        public OperationFailedException Exception => _lazyException.Value;
+
+        /// <summary>
+        /// Retrieves the result of the operation, throwing an exception if the operation failed or hasn't completed.
+        /// Unlike <see cref="Task{T}.Result"/>, this does not block.
+        /// </summary>
+        /// <remarks>
+        /// Only the in-memory representation of the operation (this object) is consulted for its state.
+        /// </remarks>
+        /// <exception cref="OperationFailedException">The operation completed with an error.</exception>
+        /// <exception cref="InvalidOperationException">The operation has not completed yet.</exception>
+        public T Result
+        {
+            get
+            {
+                switch (RpcMessage.ResultCase)
+                {
+                    case Operation.ResultOneofCase.Error:
+                        throw Exception;
+                    case Operation.ResultOneofCase.Response:
+                        return RpcMessage.Response.Unpack<T>();
+                    default:
+                        throw new InvalidOperationException();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Polls the operation until it is complete, returning the completed operation.
+        /// </summary>
+        /// <param name="pollSettings">The settings to use for repeated polling, or null
+        /// to use <see cref="PollSettings.Default"/>.</param>
+        /// <remarks>
+        /// If this object already represents a completed operation, it is returned immediately,
+        /// with no further RPCs.
+        /// </remarks>
+        /// <returns>The completed operation, which can then be checked for errors or a result.</returns>
+        public Operation<T> PollUntilCompleted(PollSettings pollSettings = null)
```

Comment:
The rationale is to match IsCompleted, which in turn matches Task.IsCompleted.
(I think an early version did in fact have PollUntilComplete...)